src/aiodynamo/http/httpx.py

```diff
@@ -1,19 +1,38 @@
 import json
 from dataclasses import dataclass
+from functools import wraps
 from typing import Any, Dict, Optional
 
 from aiodynamo.types import Timeout
-from httpx import AsyncClient
+from httpx import AsyncClient, HTTPError
```

Comment:
I wonder if `httpx` passes underlying socket errors through

Reply:
judging solely based on the names of the exceptions, it looks like it https://github.com/encode/httpx/blob/master/httpx/_exceptions.py

Reply:
I guess we too may need something like this:
https://github.com/encode/httpx/blob/53804173bb734ce94d13d1b34c0ebaab46abd8c3/tests/dispatch/test_http2.py#L155